matrix/stdlib_imports.hpp

```diff
@@ -0,0 +1,130 @@
+/**
+ * @file stdlib_imports.hpp
+ *
+ * This file is needed to shadow the C standard library math functions with ones provided by the C++ standard library.
+ * This way we can guarantee that unwanted functions from the C library will never creep back in unexpectedly.
+ *
+ * @author Pavel Kirienko <pavel.kirienko@zubax.com>
+ */
+
+#pragma once
+
+#include <cmath>
+#include <cstdlib>
+#include <cinttypes>
+
+namespace matrix {
+
+#if defined(__PX4_NUTTX)
```

Comment:
@pavel-kirienko

For the record no there is none. So far, a change to add __NUTTX__ to the generated config.h will not be accepted upstream. So this has to be set in the build by the flags, and we do that with __PX4_NUTTX.